load/filesystem.go

```diff
@@ -0,0 +1,75 @@
+package load
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+var (
+	ErrWritingToFile   = errors.New("error writing to file")
+	ErrInvalidSavePath = errors.New("invalid save path")
+)
+
+const filePermission = 0o600
+
+type FileSystem interface {
+	Write(path string, data []byte) error
+	Read(path string) ([]byte, error)
+	NormalizeSuffix(path string) string
+}
+
+type LocalFileSystem struct {
```

Comment:
I'm skepitcal we need this abstraction, the go stdlib has plenty of good filesystem io primitives and if anything adding this will constrain users of this code.

Reply:
fair enough, I was looking at this with myopically for the load package only. I'll revert and add some new tests

Reply:
@tmc I've removed the io package abstraction. There is a minimum set of prompt types supported with this PR, which doesn't include chatprompttemplate . It has a more complex interface structure.  It's probably not suitable to leave an unimplemented function in the code . I can close the PR if this is not desirable

```
func (p ChatPromptTemplate) Save(path string) error {
	_ = path
	return ErrNotImplemented
}
```